src/react_native/navigation.cljs

```diff
@@ -89,3 +89,8 @@
          (reset! constants {:top-bar-height     (.-topBarHeight consts)
                             :bottom-tabs-height (.-bottomTabsHeight consts)
                             :status-bar-height  (.-statusBarHeight consts)})))
+
+(defn bind-component
+  [^js/Object this component-id]
+  (set! (. this -navigationEventListener)
+    (.. Navigation events (bindComponent this component-id))))
```

Comment:
This function allow us to create a react class and use the react-native-navigation lifecycle methods to detect when a component has appeared or disappeared from user's screen